Controller/TwigExceptionController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\TemplateReference;
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Log\DebugLoggerInterface;
+
+/**
+ * Custom ExceptionController that uses the view layer and supports HTTP response status code mapping.
+ * It additionally is able to prepare the template parameters for the core EngineInterface.
+ */
+class TwigExceptionController extends ExceptionController
+{
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    public function setTemplating(EngineInterface $templating)
+    {
+        $this->templating = $templating;
```

Comment:
I remember one thing we talked about in #1171, but for which either did not come to a conclusion or I missed to implement that: Should we add special handling if the user doesn't have the templating component but only wants to use the `Twig_Environment` class?